hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/NearCachedClientCacheProxy.java

```diff
@@ -231,39 +231,49 @@ protected void onPutIfAbsentAsyncInternal(K key, V value, Data keyData, Data val
     @Override
     protected void getAllInternal(Set<? extends K> keys, Collection<Data> dataKeys, ExpiryPolicy expiryPolicy,
                                   List<Object> resultingKeyValuePairs, long startNanos) {
-        Map<Object, Data> keyMap = null;
         if (serializeKeys) {
-            keyMap = toDataKeys(keys, dataKeys);
+            toDataKeysWithReservations(keys, dataKeys, null, null);
         }
-        Collection<?> ncKeys = serializeKeys ? dataKeys : new ArrayList<Object>(keys);
+        Collection<?> ncKeys = serializeKeys ? dataKeys : new ArrayList<K>(keys);
```

Comment:
Good catch! That was just a hasty fix before, since I realized that modifying the user given collection is a very bad idea (e.g. could be unmodifiable)... I'll update the PR.